gadget/update.go

```diff
@@ -132,3 +207,82 @@ func canUpdateVolume(from *PositionedVolume, to *PositionedVolume) error {
 	}
 	return nil
 }
+
+type updatePair struct {
+	from *PositionedStructure
+	to   *PositionedStructure
+}
+
+func resolveUpdate(oldVol *PositionedVolume, newVol *PositionedVolume) (updates []updatePair) {
+	for j, oldStruct := range oldVol.PositionedStructure {
+		newStruct := newVol.PositionedStructure[j]
+		if newStruct.Update.Edition != oldStruct.Update.Edition {
```

Comment:
In that case, the code It defaults to edition 0. I discussed this a bit with @mvo5 and the conclusion was we should enforce that the new edition higher or equal than the old one. Once added, it would not be removed.

Reply:
The check has been updated as we discussed. Updates are only picked up if `edition` is larger than the old one.